src/meteors/shap/shap.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import annotations
+
+import shap
+import numpy as np
+import pandas as pd
+import torch
+
+from .explanation import SHAPExplanation, AVAILABLE_SHAP_EXPLAINERS
+
+from meteors.models import ExplainableModel
+
+from typing import Literal
+
+
+class HyperSHAP:
+    def __init__(
+        self,
+        callable: ExplainableModel,
+        masker,
+        explainer_type: Literal["exact", "tree", "kernel", "linear"] | None = None,
+        **kwargs,
+    ) -> None:
+        if not isinstance(callable, ExplainableModel):
+            raise TypeError(f"Expected ExplainableModel as callable, but got {type(callable)}")
+
+        self.explainable_model = callable
+
+        self.explainer_type = explainer_type.lower() if explainer_type is not None else None  # type: ignore
+
+        if self.explainer_type not in AVAILABLE_SHAP_EXPLAINERS and self.explainer_type is not None:
+            raise ValueError(
+                f"Invalid explainer type: {explainer_type}. Available options: {AVAILABLE_SHAP_EXPLAINERS}"
+            )
+
+        try:
+            if self.explainer_type is None or self.explainer_type == "exact":
+                self._explainer = shap.Explainer(self.explainable_model.forward_func, masker, **kwargs)
+            elif self.explainer_type == "tree":
+                self._explainer = shap.TreeExplainer(self.explainable_model.forward_func, masker, **kwargs)
+            elif self.explainer_type == "kernel":
+                self._explainer = shap.KernelExplainer(self.explainable_model.forward_func, masker, **kwargs)
+            elif self.explainer_type == "linear":
+                self._explainer = shap.LinearExplainer(self.explainable_model.forward_func, masker, **kwargs)
```

Comment:
What about the else?

Reply:
there is an if above that validates if `explainer_type` is correct. Thus, else block is not required.